rts/src/gc.c

```diff
@@ -0,0 +1,234 @@
+#include <assert.h>
+#include <stdlib.h>
+
+#include "tim.h"
+#include "mem.h"
+#include "dump.h"
+#include "gc.h"
+
+#ifndef NO_GC
+
+/*****************/
+/* Mark handling */
+/*****************/
+Mark gc_mark=1;
+
+void mark_refresh() {
+  gc_mark = (gc_mark + 1) % 100;
+  if (gc_mark <= 0) {gc_mark = 1;}
+}
+
+/**********************/
+/* location recording */
+/**********************/
+
+gc_data data_locations;
```

Comment:
Is the `data` here related to data constructors / data frames in any way? 

If not, perhaps we could rename it to avoid confusion.